### PR TITLE
WIP/ DOC: Move 'For Developers' content from wiki to contributing docs. #30232

### PR DESCRIPTION
**Pull request by souvik3333**

### CODE REVIEW

**Review by gfyoung — CHANGES_REQUESTED** [review]
There is a noticeable amount of Python 2 discussion here, even though we don't support Python 2 anymore in the codebase.

Can you explain why you added so much content about Python 2 compatibility?

**Comment by datapythonista** [comment]
Thanks @souvik3333, I think moving things from the wiki here is a good idea, but looks like the page you moved is not relevant anymore, now that we don't support Python 2.

Closing, but we should remove the page from the wiki (I don't have permissions, or may be it's just that it's not possible in the mobile version).

**Comment by souvik3333** [comment]
> Can you explain why you added so much content about Python 2 compatibility?

I was working on issue #30232 and this content was mentioned there. @gfyoung @datapythonista Can you verify about the other content in the list?

**Comment by alimcmaster1** [comment]
https://github.com/pandas-dev/pandas/issues/30410 @souvik3333 there is some discussion going on about it here

**Comment by souvik3333** [comment]
> #30410 @souvik3333 there is some discussion going on about it here

Yes, I just found it. Thank you.

**Comment by simonjayhawkins** [comment]
> Thanks @souvik3333, I think moving things from the wiki here is a good idea, but looks like the page you moved is not relevant anymore, now that we don't support Python 2.

I think we want the `Imports (aim for absolute)` section and maybe more.

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -569,8 +569,40 @@ do not make sudden changes to the code that could have the potential to break
...
 
-Additional standards are outlined on the `code style wiki
-page <https://github.com/pandas-dev/pandas/wiki/Code-Style-and-Conventions>`_.
+Cross-compatible code
+---------------------
+
+It's important to write code that will be compatible with most recent version of
+Python 3.
+
+Python 2/3 Compatibility
```
Comment (simonjayhawkins) [inline doc/source/development/contributing.rst]: only py3.6+ is supported, so I think we can remove this too

**Comment by simonjayhawkins** [comment]
@souvik3333 I think the additions here should be added to doc/source/development/contributing_code_guide.rst once #30129 is merged

**Comment by datapythonista** [comment]
Closing this, since all the Python 2/3 documentation is not useful anymore. @souvik3333 as said, once #30129 is merged, you can add a section about the imports there. Thanks!

**Pull request closed without merging**